MAVProxy/modules/mavproxy_steamvr.py

```diff
@@ -0,0 +1,303 @@
+
+                # send origin at 1Hz
+                mav.mav.set_gps_global_origin_send(self.target_system,
+                                                   int(self.vr_settings.origin_lat*1.0e7),
+                                                   int(self.vr_settings.origin_lon*1.0e7),
+                                                   int(self.vr_settings.origin_alt*1.0e3),
+                                                   time_us)
+                last_origin_send = now
+
+            if self.vr_settings.gps_rate > 0:
+                gps_period = 1 / self.vr_settings.gps_rate
+                if now - last_gps_send > gps_period:
+                    '''send GPS data at the specified rate, trying to align on the given period'''
+                    yaw_cd = int(math.degrees(yaw) * 100)
+                    if yaw_cd == 0:
+                        # the yaw extension to GPS_INPUT uses 0 as no yaw support
+                        yaw_cd = 36000
+
+                    gps_lat, gps_lon = mavextra.gps_offset(self.vr_settings.origin_lat,
+                                                        self.vr_settings.origin_lon,
+                                                        pos_ned.y, pos_ned.x)
+                    gps_alt = self.vr_settings.origin_alt - pos_ned.z
+
+                    gps_week, gps_week_ms = get_gps_time(now)
+
+                    if self.vr_settings.gps_nsats >= 6:
+                        fix_type = 3
+                    else:
+                        fix_type = 1
+                    mav.mav.gps_input_send(time_us, 0, 0, gps_week_ms, gps_week, fix_type,
+                                        int(gps_lat*1.0e7), int(gps_lon*1.0e7), gps_alt,
+                                        1.0, 1.0,
+                                        vel_ned.x, vel_ned.y, vel_ned.z,
+                                        0.2, 1.0, 1.0,
+                                        self.vr_settings.gps_nsats,
+                                        yaw_cd)
+                    last_gps_send = now
+                    gps_frame_count += 1
+
+            if self.vr_settings.vision_rate > 0:
+                # send VISION_POSITION_ESTIMATE
+                # we force mavlink1 to avoid the covariances which seem to make the packets too large
+                # for the mavesp8266 wifi bridge
+                mav.mav.global_vision_position_estimate_send(time_us,
+                                                            pos_ned.x, pos_ned.y, pos_ned.z,
+                                                            roll, pitch, yaw, force_mavlink1=True)
+                vision_frame_count += 1
+
+                if self.vr_settings.vision_speed:
+                    # send VISION_SPEED_ESTIMATE
+                    mav.mav.vision_speed_estimate_send(time_us, vel_ned.x, vel_ned.y, vel_ned.z, force_mavlink1=True)
+
+    def cmd_start(self):
```

Comment:
Add something user-friendly if openvr fails to start, like "Unable to start OpenVR. Check OpenVR is installed and configured".

Reply:
I have added a try: exception: round this, it seems to work but I'm not sure it that is the best way